internal/gpulib/device.go

```diff
@@ -0,0 +1,126 @@
+/**
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package gpulib
+
+import (
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
```

Comment:
I would use the nvml wrapper in go-nvlib instead of this one directly since that is the one we will need to rely on once this is pulled into go-nvlib.

Reply:
Yes, I was wondering about that. I suppse that we're missing some functionality from that API (e.g. the NVLINK functions) but we should still be able to address that using embedding and only add the new functions locally.